File: tests/property/python/areas/lwe.py
"""SageMath side of the ``lwe`` property-test area.

The cases compare deterministic structural facts.  The sample-vector case
constructs and calls the real oracle, but compares only the vector dimension
because SageMath and sagemath-ts intentionally use different random streams.

Cases: tests/property/cases/lwe.cases.json
"""

from sage.crypto.lwe import LindnerPeikert, Regev, UniformNoiseLWE


def regev_q(n):
    return Regev(n).K.order()


def regev_dimension(n):
    return Regev(n).n


def lindner_peikert_m(n):
    return LindnerPeikert(n).m


def sample_vector_length(n):
    a, _ = Regev(n)()
    return len(a)


def uniform_noise_min_n(n):
    UniformNoiseLWE(n)
    return 'valid'


FUNCTIONS = {
    'regev_q': regev_q,
    'regev_dimension': regev_dimension,
    'lindner_peikert_m': lindner_peikert_m,
    'sample_vector_length': sample_vector_length,
    'uniform_noise_min_n': uniform_noise_min_n,
}

File: tests/property/typescript/areas/lwe.ts
/**
 * sagemath-ts side of the `lwe` property-test area.
 *
 * Cases: tests/property/cases/lwe.cases.json
 * SageMath counterpart: tests/property/python/areas/lwe.py
 */

import {
  LindnerPeikert,
  Regev,
  UniformNoiseLWE,
} from '../../../../packages/sagemath-ts/src/crypto/lwe.js';

export const functions = {
  regev_q: (n: bigint): bigint => new Regev(n).K.order,
  regev_dimension: (n: bigint): number => new Regev(n).n,
  lindner_peikert_m: (n: bigint): number | null => new LindnerPeikert(n).m,
  sample_vector_length: (n: bigint): number => new Regev(n).call()[0].length,
  uniform_noise_min_n: (n: bigint): string => {
    new UniformNoiseLWE(n);
    return 'valid';
  },
};
